Enabled header_saveload feature

VerevkinAlexander opened the pull request:
Closes #86 
I enabled header_saveload feature

![image](https://user-images.githubusercontent.com/43240282/55411910-2448ea80-556f-11e9-80c3-4d4dfcbe29a8.png)

VerevkinAlexander committed:
Enabled header_saveload feature

sschiessl-bcp commented:
The cloud save and load feature works, but displays many other layouts
![image](https://user-images.githubusercontent.com/33128181/55616612-79b11180-5792-11e9-82be-f4bcf3ccc416.png)

Where do they come from?

sschiessl-bcp commented:
When I save a chart I see this additional selector
![image](https://user-images.githubusercontent.com/33128181/55616664-99e0d080-5792-11e9-9aef-3583de54aa04.png)

But when I then reload the website, that additional selector is gone. Where does it come from?

VerevkinAlexander commented:
> Where do they come from?

They come from other users those use public user id
https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
We can create own or add to the settings store possibility to add own exist account for each user
https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access

VerevkinAlexander commented:
> that additional selector is gone. Where does it come from?

It caused because we initialize tradeview widget each time when we mount component

sschiessl-bcp commented:
> > Where do they come from?
> 
> They come from other users those use public user id
> https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
> We can create own or add to the settings store possibility to add own exist account for each user
> https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access

First step I would like to a user to only see templates saved by himself

sschiessl-bcp commented:
> > > Where do they come from?
> > 
> > 
> > They come from other users those use public user id
> > https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
> > We can create own or add to the settings store possibility to add own exist account for each user
> > https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access
> 
> First step I would like to a user to only see templates saved by himself

Is this implemented?

VerevkinAlexander commented:
Not yet

VerevkinAlexander commented:
there are three ways to implement it: 
- save data to Local storage
- save data to own data storage https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#using-high-level-saveload
- save data to tiny storage provided by http://saveload.tradingview.com (unsafe) https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#using-demo-charts-and-study-templates-storage

VerevkinAlexander commented:
Which way should I choose?

VerevkinAlexander committed:
Added save and load charts into memory

VerevkinAlexander commented:
I`ve added save /load charts functionality please look.  I`ve temporarily used  in memory charts saving

sschiessl-bcp commented:
LocalStorage would be preferable

sschiessl-bcp commented:
Accidently closed

VerevkinAlexander committed:
added saving to the local storage

VerevkinAlexander committed:
added close modal after last chart layout deleting

VerevkinAlexander commented:
I`ve implemented in local storage saving and charts deleting. Please review it

VerevkinAlexander commented:
I think what it will be cool if users have the possibility to share their charts layout. What do you think about it?

sschiessl-bcp commented:
> I think what it will be cool if users have the possibility to share their charts layout. What do you think about it?

I agree, in what way would you propose this to be done?

VerevkinAlexander commented:
We can add export via JSON file uploading also it can be coded in base64 format. I wonder UI will have 2 buttons (expor and import) on the Load charts layout modal 
![image](https://user-images.githubusercontent.com/43240282/59346372-ce38b600-8d1a-11e9-9576-d03621d9725b.png)

sschiessl-bcp commented on app/components/Exchange/TradingViewPriceChart.jsx:
```diff
@@ -147,18 +176,95 @@ export default class TradingViewPriceChart extends React.Component {
+            state.showSaveModal !== this.state.showSaveModal ||
+            np.chartHeight !== this.props.chartHeight ||
+            this.props.charts.size !== np.charts.size ||
+            this.tvWidget ||
```
Please double check the logic when refactoring and adjust the mistake here

VerevkinAlexander replied:
sorry, fixed

sschiessl-bcp commented:
Please rebase

VerevkinAlexander committed:
fixed update logic

sschiessl-bcp commented:
Accepted and forwarded

VerevkinAlexander commented:
Rebased to newest develop
Plz review it

VerevkinAlexander committed:
- Added same name error
- refactored charts header
- Added saving charts accorf=ding to market
- Implemented last enabled chart loading

VerevkinAlexander commented:
fixed issues https://github.com/bitshares/bitshares-ui/pull/2804#issuecomment-506967445

VerevkinAlexander committed:
-added translation code

sschiessl-bcp commented on app/components/Exchange/Exchange.jsx:
```diff
@@ -3072,7 +3075,7 @@ class Exchange extends React.Component {
                     className={cnames(
                         centerContainerWidth > 1200 &&
-                        (verticalOrderBook || verticalOrderBook)
+                            (verticalOrderBook || verticalOrderBook)
```
Whole Exchange file is reformat only, please revert (rebase if necessary)

VerevkinAlexander replied:
I`ve changed https://github.com/blockchainprojects/bitshares-ui/pull/89/files/9ac96a83c547065d6e303fcbe16c8306eabb39ee#diff-3aaa4fc021c75bd17e1e59adb802e56dL2753 
Other were changed by pre-commit hook

VerevkinAlexander committed:
Revert Exchange pre-commit changes

Commit:
switch right depending on chart

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

sschiessl-bcp commented:
?

VerevkinAlexander commented:
> ?

I`ve updated develop and looks like it merged automatically because it has been merged to the origin develop
